src/app/api/datasets/route.ts

```diff
@@ -0,0 +1,21 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { datasetList } from '@/services/discovery';
+import { ExtendedSession, authOptions } from '@/utils/auth';
+import { getServerSession } from 'next-auth';
+import { NextResponse } from 'next/server';
+
+export async function POST(request: Request) {
```

Comment:
Where are you calling this endpoint?

I saw that you are directly calling datasetList [here](https://github.com/GenomicDataInfrastructure/gdi-userportal-frontend/pull/190/files#diff-4890f1283a8eac430519283a837534d5824c61b6d9b24bde2575f75df47f94cdR18)

So frontend seems to be calling dataset discovery directly no? It should have been calling this point instead